src/main/java/com/tudu/todoapp/rest/controllers/ToDoItemController.java

```diff
@@ -0,0 +1,85 @@
+package com.tudu.todoapp.rest.controllers;
+
+import com.tudu.todoapp.entities.TodoItem;
+import com.tudu.todoapp.repositories.TodoItemRepository;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/todoitems")
+public class ToDoItemController {
+
+    private final TodoItemRepository todoItemRepository;
+
+    @Autowired
+    ToDoItemController(TodoItemRepository todoItemRepository){
+        this.todoItemRepository = todoItemRepository;
+    }
+
+    @GetMapping
+    public ResponseEntity<Page<TodoItem>> getAllToDoItems(
```

Comment:
Should a page or a list be returned here idk, you can call `toList()` on a `Page` object I think.

Reply:
I was not sure as well, but I can change it to List if that is okay with you.